app/controllers/UserController.java

```diff
@@ -30,12 +30,15 @@
     public Result register() {
         User user = Json.fromJson(request().body().asJson(), User.class);
 
+        if(!userHelper.isSetForRegistration(user)){
```

Comment:
user.isValid and response message for validation fail should always be the same (validation means request validation: format, missing data, etc)